stl/inc/format

```diff
@@ -343,72 +337,391 @@ public:
+                _Arg_index, basic_format_arg<_Context>::handle(_Val));
+        }
+    }
+
+    void _Store(const size_t _Arg_index, float _Val) noexcept {
+        _Store_impl<float, _Basic_format_arg_type::_Float_type>(_Arg_index, _Val);
+    }
+
+    void _Store(const size_t _Arg_index, double _Val) noexcept {
+        _Store_impl<double, _Basic_format_arg_type::_Double_type>(_Arg_index, _Val);
+    }
+
+    void _Store(const size_t _Arg_index, long double _Val) noexcept {
+        _Store_impl<long double, _Basic_format_arg_type::_Long_double_type>(_Arg_index, _Val);
+    }
 
-    using _Value_type = basic_format_arg<_Context>;
+    void _Store(const size_t _Arg_index, const _CharType* _Val) noexcept {
+        _Store_impl<const _CharType*, _Basic_format_arg_type::_CString_type>(_Arg_index, _Val);
+    }
+
+    template <class _Traits>
+    void _Store(const size_t _Arg_index, basic_string_view<_CharType, _Traits> _Val) noexcept {
+        _Store_impl<basic_string_view<_CharType>, _Basic_format_arg_type::_String_type>(
+            _Arg_index, basic_string_view<_CharType>{_Val});
+    }
+
+    template <class _Traits, class _Alloc>
+    void _Store(const size_t _Arg_index, basic_string<_CharType, _Traits, _Alloc> _Val) noexcept {
+        _Store_impl<basic_string_view<_CharType>, _Basic_format_arg_type::_String_type>(
+            _Arg_index, _Basic_format_arg_type::_String_type, basic_string_view<_CharType>{_Val.data(), _Val.size()});
+    }
+
+    void _Store(const size_t _Arg_index, nullptr_t) noexcept {
+        _Store_impl<const void*, _Basic_format_arg_type::_Pointer_type>(_Arg_index, static_cast<const void*>(nullptr));
+    }
+
+    // clang-format off
+    template <class _Ty>
+        requires is_void_v<_Ty>
+    void _Store(const size_t _Arg_index, const _Ty* p) noexcept {
+        // clang-format on
+        _Store_impl<const void*, _Basic_format_arg_type::_Pointer_type>(_Arg_index, static_cast<const void*>(p));
+    }
 };
 
-// TODO: test coverage
+template <class _Context = format_context, class... _Args>
+_Format_arg_store<_Context, _Args...> make_format_args(const _Args&... args) {
+    _Format_arg_store<_Context, _Args...> _Arg_store{};
+    size_t _Arg_index = 0;
+    (_Arg_store._Store(_Arg_index++, args), ...);
+    return _Arg_store;
+}
+
+template <class _Context = wformat_context, class... _Args>
+_Format_arg_store<_Context, _Args...> make_wformat_args(const _Args&... args) {
+    _Format_arg_store<_Context, _Args...> _Arg_store{};
+    size_t _Arg_index = 0;
+    (_Arg_store._Store(_Arg_index++, args), ...);
+    return _Arg_store;
+}
+
 template <class _Context>
 class basic_format_args {
+private:
+    using _CharType = typename _Context::char_type;
+
+    template <class _Ty>
+    _NODISCARD static constexpr auto _Get_value_from_memory(const byte* _Storage) noexcept {
+        byte _Temp[sizeof(_Ty)];
+        _Copy_unchecked(_Storage, _Storage + sizeof(_Ty), _Temp);
+        return _Bit_cast<_Ty>(_Temp);

```

Comment:
Let's eliminate a copy here so the optimizer doesn't have to:
```suggestion
        auto& _Temp = *reinterpret_cast<const byte(*)[sizeof(_Ty)]>(_Storage);
        return _Bit_cast<_Ty>(_Temp);
```

Reply:
This was me trying to write it so that we wont get too much trouble when format becomes constexpr, but now I think this was a bad idea